src/sd.jl

```diff
@@ -167,6 +203,25 @@ function build_variable(_error::Function, variables::Matrix{<:ScalarVariable}, :
     return VariablesConstrainedOnCreation(_vectorize_variables(_error, variables), set, shape)
 end
 
+"""
+    build_constraint(_error::Function, variables, ::SkewSymMatrixSpace)
+
+Return a `VariablesConstrainedOnCreation` of shape [`SkewSymMatrixShape`](@ref)
+creating variables in `MOI.Reals`, i.e. "free" variables unless they are
+constrained after their creation.
+
+This function is used by the [`@variable`](@ref) macro as follows:
+```julia
+@variable(model, Q[1:2, 1:2] in SkewSymMatrixSpace())
+```
+"""
+function build_variable(_error::Function, variables::Matrix{<:ScalarVariable}, ::SkewSymMatrixSpace)
+    n = _square_side(_error, variables)
+    set = MOI.Reals(div(n^2 - n, 2))
+    shape = SkewSymmetricMatrixShape(n)
+    return VariablesConstrainedOnCreation(_vectorize_variables(_error, variables), set, shape)
```

Comment:
`vectorize_variables` also include the diagonal, this why the test are failing. I would guess because of this, you add to many variables so test the number of variables added

Reply:
ok